app/src/main/java/com/samco/trackandgraph/antlr/evaluation/EvaluationErrors.kt

```diff
@@ -0,0 +1,57 @@
+package com.samco.trackandgraph.antlr.evaluation
+
+import com.samco.trackandgraph.antlr.ast.Error
+import com.samco.trackandgraph.antlr.ast.Point
+import com.samco.trackandgraph.antlr.ast.Position
+
+private fun emptyPosition() : Position {
+    return Position(Point(0, 0), Point(0, 0))
+}
+
+fun throwUnsupportedArgumentTypeError(functionName: String, main: Value, other: Value, allowedTypes: String) : Nothing {
+    throw UnsupportedArgumentTypeError(
+        functionName,
+        listOf(main::class.simpleName!!, other::class.simpleName!!),
+        allowedTypes, emptyPosition()
+    )
+
+}
+
+class DoubleDeclarationError(val varName: String, position: Position) :
+    Error("Variable '$varName' was already declared!", position)
```

Comment:
Can't we just localize the `"Variable '$varName' was already declared!"` bit and keep the `$varName` in all versions? We just need to replace that string with a getString(R.string.errorDoubleDeclaration) or do we need more? But I didn't really think about that yet, good point, I'll check if it could work as I just described it.

Also we are using a custom Error class here (called Error, terrible name, but that was the name in the original code) so the position doesn't get concatenated there.

Reply:
And i'll rename the function